### ISSUE TITLE
Incompatible datapack (with Pictures)

### ISSUE THREAD

**Issue by Coraline77447** [CreativeMD/EnhancedVisuals#58]
Hi, it's me from the comments on EnhancedVisuals. Here's the pictures of what's been  going on with the datapacks of; EnhancedVisuals, CreativeCore, and ItemPhysic+Full:

EnhancedVisuals:
![16064092754775823060270723402383](https://user-images.githubusercontent.com/72243304/100376461-cdf2b000-2fcc-11eb-9115-7871414860e0.jpg)

![1606409452834120837204453215091](https://user-images.githubusercontent.com/72243304/100376549-f24e8c80-2fcc-11eb-9ad9-37ddba65173b.jpg)

CreativeCore:
![16064094953805956009264560504105](https://user-images.githubusercontent.com/72243304/100376611-1611d280-2fcd-11eb-9647-04d977cb0d9d.jpg)

![16064095573118341284600555821726](https://user-images.githubusercontent.com/72243304/100376675-2c1f9300-2fcd-11eb-99f7-60d780806ca3.jpg)

ItemPhysic+Full:
![16064096108298670813890666544499](https://user-images.githubusercontent.com/72243304/100376748-48bbcb00-2fcd-11eb-8ae6-588b072e8699.jpg)

![16064096431477133957799164448068](https://user-images.githubusercontent.com/72243304/100376792-5bce9b00-2fcd-11eb-9d93-2217f63a99c4.jpg)

**Comment by CreativeMD** [comment]
Oh, that's not how you install mods :D you can't do that. You should probably watch a tutorial about how to install forge. The mods need to be in the mods folder.

**Comment by Coraline77447** [comment]
I have them in my mods folder, these pictures are to show you the mods datapack where their pack_format is within their datapacks and they're outdated which is why you see them in red boxes instead of gray boxes like all of the other mods data packs,  when I'd hovered my mouse over then they say; 

"Incompatible
(Made for an older version of Minecraft)

But instead of the mods themselves being outdated its their datapacks, even tho it says made for an older version of Minecraft what it really means is this "(Made for an  older version of minecraft's  datapack) since this problem resides  in the datapacks not the mods themselves or the game itself. Mc's datapack is at version 6 while your mods datapacks are currently at version 4 and I can't manually update their meta packs to version 6 so you gotta update them to the current version of mc's datapack. I know that EnhancedVisuals, ItemPhysic+Full, and CreativeCore doesn't have anything to do with datapacks whatsoever but if the game along side with forge can see that your mods have datapacks that your mods are using for their resources (resource packs, and other stuff that makes the mod work correctly or in a certain way) then the game will tell us that the mods have datapacks and if the game sees that the mods datapack is outdated then the game will warn us that the mods datapack is outdated and needs to be updated or else you'll not be able to join a world with an outdated datapack, also depending on a mod that has a certain datapack that's outdated, that outdated datapack can potentially harm mc's datapack system and the world's datapack to the point that it damages the game's datapack and the world that you made or trying to make! I can't even make a new modded world anymore cause of how damaged my game's datapack is and there's no way to fix mc's datapack unless there's a major update like the Caves & Cliffs update.

**Comment by CreativeMD** [comment]
Increase the version to 6, but actually I have no idea why this could be important.

**Comment by Coraline77447** [comment]
Thank you for that also I would like to apologize for the confusion and all I may have caused. All that I was trying to ask was for you to update your mod's datapacks. I've went out of my way to try to make all 25 mods that I'd played with that had/has this problem. For some of those 25 mods I was able to go to their pack.meta where their pack_format is and was able to make the adjustments while there's some mods that I can't simply do that cause in their pack.meta file there's a line underneath pack_format called "_comment" which I'm guessing that I can't manually make the adjustments myself cause the owners of their mods have set the meta version to what they see fit while MC's meta version is at 6. If I were to have made those adjustments while the _comment suggests say v4 then I could've potentially caused some damage, alot of damage or worse to those mods like yours since your mods has that line underneath pack_format. I don't want to harm them so that's why I reported this issue to those mods and 2 out of 17 of the remaining mods were your mods CreativeCore and ItemPhysic+Full. Now there's 15 that still needs their pack_format to be updated and unfortunately EnhancedVisuals is one of them but I can live without EnhancedVisuals. When you do manage to update EnhancedVisuals's pack_format I'll have a separate modded world from the one I'm currently playing on to spice things up so that when I get bored of my current modded world I'll make a new one with a different set of mods that'll incloude EnhancedVisuals (When it's resource pack version has been updated). I also have no idea why Mojang implemented this. Maybe that way people can make a datapack for the mod(s) they're playing with as long as the mod is capable of running with a custom datapack that was made for it? Or to make modding the game a bit easier by making a datapack that adds custom blocks and items and even custom behaviors for entities/mobs? (Btw sorry for my long reply and for how long all of my replies and my comment at the beginning was.)

**Comment by CreativeMD** [comment]
Updated it for EnhancedVisuals as well.

Just keep in mind that you cannot modify any mod which is currently used by minecraft. Just close the game and you can edit everything as you want to.

**Comment by Coraline77447** [comment]
I always edit mods outside of the game unless the game allows for a live in-game edit in the mod's config file or in-game config while the game is running. Unless the owner of such a mod decides to remove that ability to only allow to edit in it's config file outside of the game. But for the mod's pack_format (packmc.meta folder) is only allowed to be edited outside of the game. I know that I can't edit a mod's pack_format while in the game cause that may cause major issues and to cause the game to crash cause the mod's pack_format was changed while the game was running and doesn't recognize the change properly and wouldn't know what to do about the changes so it's only option is to shut itself down/forcefully stop itself from running. Then we would have to restart the game so that it can recognize the changes. I don't want to be rude or anything but I do know what I'm doing as of playing with the mods that I have installed in my mods folder and I understand how they work (Mostly). I know what consequences would be if I were to make a wrongful change to them but if I don't know what the consequences would be then the only way to know is by trial and error in order to succeed on making the correct change by launching the game and then the game telling me what needs to be fixed unless the game crashes. I don't quite understand crash repots but thanks to a mod (Blame mod) I'll figure out which mod caused the crash to happen so that I can report my findings to the owner or owners of multiple different mods at once. I know the consequences of changing a mod's pack_format when there's a _comment line with a note that says what the mod can only use and by changing it may cause some problems and I would change it back to the way they were just like how I'd changed Quark's pack_format when it's _comment says it needed to be at v4 when the pack_format was already at v5 for some odd reason (the owner set it to v5 not me even tho it's _comment w/note says it needs to be at v4) and the only way to find out if my change from a 5 to a 6 would cause something bad to happen to Quark and if so then I can simply redownload Quark and never touch it's pack_format ever again and will teach me not to do that with other mods that has _comment w/note ever again. But I'm not putting quark in my mods folder to see if my changes would work since I'm still playing in my current modded world that already has too many loaded chunks in certain directions or even all directions and I'm not willing to go thousands and thousands of blocks away in any direction just to see what Quark adds to the game regarding to worldgen. But when I make another modded world I'll have Quark in to see if the changes to it's pack_format would work.
